### ISSUE TITLE
Decimal Array Column type creates migration repeatedly

### ISSUE THREAD

**Issue by akdeb** [typeorm/typeorm#8106]
## Issue Description
I have a column decorator of the form:
```
@Column({
type: 'decimal',
array: true,
precision: 100,
scale: 50,
})
numberArray!: number[];
```

When I generate a migration for this field, I get a repeated migration of the form. This is generated repeatedly, every time I generate a new migration. 
```
import {MigrationInterface, QueryRunner} from "typeorm";

export class sample1629695051497 implements MigrationInterface {
    name = 'sample1629695051497'

    public async up(queryRunner: QueryRunner): Promise<void> {
        await queryRunner.query(`ALTER TABLE "field_value" ALTER COLUMN "numberArrayValue" TYPE numeric(100,50) array`);
        await queryRunner.query(`COMMENT ON COLUMN "field_value"."numberArrayValue" IS NULL`);
        await queryRunner.query(`ALTER TABLE "default_field_value" ALTER COLUMN "numberArrayValue" TYPE numeric(100,50) array`);
        await queryRunner.query(`COMMENT ON COLUMN "default_field_value"."numberArrayValue" IS NULL`);
    }

    public async down(queryRunner: QueryRunner): Promise<void> {
        await queryRunner.query(`COMMENT ON COLUMN "default_field_value"."numberArrayValue" IS NULL`);
        await queryRunner.query(`ALTER TABLE "default_field_value" ALTER COLUMN "numberArrayValue" TYPE numeric array`);
        await queryRunner.query(`COMMENT ON COLUMN "field_value"."numberArrayValue" IS NULL`);
        await queryRunner.query(`ALTER TABLE "field_value" ALTER COLUMN "numberArrayValue" TYPE numeric array`);
    }

}
```

This issue goes away when I am not using the `type: 'decimal'`. Is there something I am 

### Expected Behavior

Ideally I'd expect this field to be of type `numeric(100, 50)[]` and when I check on postgres, I do see that but the migration repeating incessantly is the issue I am facing. Is this is a known issue? How can I fix it? The fix I have now is to just create an array of this type:

```
@Column({
    type: 'float8',
    array: true,
  })
  numberArrayOperand!: number[];
```

### Steps to Reproduce

1. Create an entity with the column decorator above.
2. Run all existing migrations.
3. Generate a new migration, this should reflect the change with the `numberArray` field you just added.
4. Now run this new migration and generate a new migration
5. You may find a new migration file gets generated repeatedly 

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  macOS Mojave Version 10.14.6        |
| Node.js version     | v12.14.1  | 
| Typescript version  | 3.8.2  | 
| TypeORM version     | 0.2.31  | 

### Relevant Database Driver(s)

| DB Type              | Reproducible  |
| ---                  | ---           |
| `aurora-data-api`    | no            |
| `aurora-data-api-pg` | no            |
| `better-sqlite3`     | no            |
| `cockroachdb`        | no            |
| `cordova`            | no            |
| `expo`               | no            |
| `mongodb`            | no            |
| `mysql`              | no            |
| `nativescript`       | no            |
| `oracle`             | no            |
| `postgres`           | yes            |
| `react-native`       | no            |
| `sap`                | no            |
| `sqlite`             | no            |
| `sqlite-abstract`    | no            |
| `sqljs`              | no            |
| `sqlserver`          | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✖️ Yes, I have the time, and I know how to start.
 - ✅ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by akdeb** [comment]
@pleerock Saw your response on other similar errors. Have you faced this issue?

**Comment by ranjan-purbey** [comment]
@akdeb This seems to have been fixed in typeorm `v0.2.33` and later. Please upgrade and see if you still have this issue
